crypto/evp/evp_enc.c

```diff
@@ -306,6 +306,12 @@ static int evp_EncryptDecryptUpdate(EVP_CIPHER_CTX *ctx,
     bl = ctx->cipher->block_size;
 
     if (ctx->cipher->flags & EVP_CIPH_FLAG_CUSTOM_CIPHER) {
+        /* Prevent accidental "final" call */
+        if (inl <= 0 && in == NULL) {
```

Comment:
Is it actually necessary for `in` to be NULL for us to process it in this way. I'm thinking that just `inl <= 0` is a good enough check. I could envisage applications providing a valid (non-NULL) pointer and effectively saying "take 0 bytes from this location".

Reply:
Hmmm...  this is one of those times where I want to say "here be dragons", and observe that EVP_*Final does call the backend cipher function with NULL and 0 exactly.  I don't have enough insight into the diverse backends to know for sure if they all behave like nothing happened for a valid pointer with zero length, or if some have side effects.  There's quite a bit of hackery in some of them, so I tend to er toward extra caution...

Reply:
I ended up changing it anyway...